instrumentation/net/http/otelhttp/wrap.go

```diff
@@ -93,3 +93,9 @@ func (w *respWriterWrapper) WriteHeader(statusCode int) {
 	w.statusCode = statusCode
 	w.ResponseWriter.WriteHeader(statusCode)
 }
+
+func (w *respWriterWrapper) Flush() {
+	if !w.wroteHeader {
+		w.WriteHeader(http.StatusOK)
```

Comment:
Technically, the only case where this could be impactful is if someone configures another handler on top of us, and this handler tries to change the status code, they won't, as it has already been set.
My first reflex was to say that the span should trie to avoid lying if possible.

But the specification goes with your opinion (for error handling, but I believe that statement is generic enough here): https://github.com/open-telemetry/opentelemetry-specification/blob/5f772ee110addc9336d2145eaa3d906a44dfda13/specification/error-handling.md

> We assume that users would prefer to lose telemetry data rather than have the library significantly change the behavior of the instrumented application.